app/forms/idv/finance_form.rb

```diff
@@ -4,49 +4,70 @@ class FinanceForm
 
     FINANCE_TYPES = [:ccn, :mortgage, :home_equity_line, :auto_loan].freeze
 
-    attr_reader :idv_params, :finance_type, :finance_account
+    FINANCE_HTML_OPTIONS = {
+      ccn: { maxlength: 8 }
+    }
 
-    validates :finance_type, :finance_account, presence: true
+    attr_reader :idv_params, :finance_type, *FINANCE_TYPES
 
-    validates :finance_account,
+    validates :finance_type, presence: true
+
+    FINANCE_TYPES.each do |type|
+      validates type, presence: true, if: ->(form) { form.finance_type == type }
+    end
+
+    validates :ccn,
               format: { with: /\A\d{8}\z/, message: I18n.t('idv.errors.invalid_ccn') },
               if: :ccn?
 
-    validate :finance_type_valid
+    validate :validate_finance_type
```

Comment:
interested to learn why you changed this name. Do you usually name validations `validate validate_x_y_z`? 

I am thinking about the default rails validations such as `validate :thing, presence: true` 

Seems like it would make sense to name custom validations similarly, like the former`validate :finance_type_valid`, but I am open to this change. More curious on your thinking than anything else.